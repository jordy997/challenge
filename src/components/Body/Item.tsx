import React from 'react'
import { Image, StyleSheet, Text, View } from 'react-native';

const Item = ({ product, createdAt, points, is_redemption, image }) => {
    return (
        <View style={styles.item}>
            <Image source={image} style={styles.image} />
            <View>
                <Text style={styles.title}>{product}</Text>
                <Text style={styles.date}>{createdAt}</Text>
            </View>
            <Text style={styles.points}>
                <Text style={{ color: is_redemption ? '#FF0000' : '#00B833' }}>
                    {is_redemption ? '-' : '+'}
                </Text>
                {points} {'  >'}
            </Text>
        </View>
    )
};

const styles = StyleSheet.create({
    item: {
        display: 'flex',
        flexDirection: 'row',
        marginHorizontal: 10,
        height: 55,
        alignItems: 'center',
        justifyContent: 'space-between',
        marginVertical: 8
    },
    title: {
        fontSize: 14,
        fontWeight: '800',
        lineHeight: 19
    },
    date: {
        fontWeight: '400',
        lineHeight: 16,
        fontSize: 12
    },
    points: {
        fontWeight: '800',
        fontSize: 16,
        lineHeight: 22,
        alignSelf: 'center',
    },
    image: {
        width: 55,
        height: 55,
        borderRadius: 10,
        backgroundColor: 'red'
    }
})

export default Item