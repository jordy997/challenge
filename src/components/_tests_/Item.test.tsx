import React from 'react'

import { shallow } from 'enzyme'
import Item from '../Body/Item'

let wrapper

describe('Item component', () => {

    beforeEach(() => {
        wrapper = shallow(
            <Item
                createdAt='2022-12-09T10:20:00.909Z'
                product='Fantastic Granite Bacon'
                points={42416}
                image='https://loremflickr.com/640/480/technics'
                is_redemption={false}
                id='3'
                onPress={() => console.log('Hello world')}
            />
        )
    })

    describe('Rendering', () => {
        it('Item should render correctly', () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
})