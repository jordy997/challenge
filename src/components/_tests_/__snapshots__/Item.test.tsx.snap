// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component Rendering Item should render correctly 1`] = `ShallowWrapper {}`;
